magpy/gui/magpy_gui.py

```diff
@@ -5131,6 +5140,53 @@ def onLoadDI(self,event):
             self.menu_p.abs_page.AnalyzeButton.Enable()
         dlg.Destroy()
 
+    def onLoadUSGS(self,event):
+        startdate = self.plotstream.ndarray[KEYLIST.index('time')][0]
+        enddate = self.plotstream.ndarray[KEYLIST.index('time')][-1]
+        starttime = num2date(startdate)
+        endtime = num2date(enddate)
+        di_db = []
+        dlg = LoadUSGSDialog(None, title='Get USGS data')
+        if dlg.ShowModal() == wx.ID_OK:
+            extension = dlg.extensionComboBox.GetValue()
+            dt = extension[0]
+            dttype = extension[2]
+            if dttype == 'd':
+                dt = int(dt)
+            if dttype == 'w':
+                dt = int(dt) * 7
+            time = starttime - timedelta(days = dt)
+            base = 'https://geomag.usgs.gov/baselines/observation.json.php?'
```

Comment:
The service supports an endtime parameter that defaults to now, so I think these requests will overlap:
    https://geomag.usgs.gov/baselines/observation.json.php

Does magpy require each day of absolutes to be separate?